numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -426,16 +440,25 @@ UNICODE_setitem(PyObject *op, void *ov, void *vap)
     char *buffer;
 #endif
 
-    if (!PyBytes_Check(op) && !PyUnicode_Check(op) &&
-            PySequence_Check(op) && PySequence_Size(op) > 0) {
+    /* Handle case of assigning from an array scalar */
+    if (PyArray_Check(op) && PyArray_NDIM((PyArrayObject *)op) == 0) {
```

Comment:
So, here we use
```
PyArray_ToScalar(PyArray_BYTES((PyArrayObject *)op),
                              (PyArrayObject *)op);
```
But in `cdouble`, [we have](https://github.com/eric-wieser/numpy/blob/633568e99a7bf992c12bbd827c7b462e50e8fc78/numpy/core/src/multiarray/arraytypes.c.src#L278-L280)
```
op2 = PyArray_DESCR((PyArrayObject *)op)->f->getitem(
                                    PyArray_BYTES((PyArrayObject *)op),
                                    (PyArrayObject *)op);
```

Are these different in any important ways?